### DISCUSSION TITLE
FIPS Compliance with COMPLIANCE_POLICY ENV variable

### DISCUSSION THREAD

**Discussion by smvdn** [istio/istio#53270] [Q&A]
We want to enable FIPS compliance with Istio.

We could see the envoy proxy being used in istio/proxyv2 is not having the [FIPS version flag](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl.html#fips-140-2). Do we need to custom build Istio Proxy Image with *--define boringssl=fips* along with adding the environment variable COMPLIANCE_POLICY="fips-140-2"?  Is there any specific FIPS compliant image tag we should use along with the env variable?

![image](https://github.com/user-attachments/assets/204e3fd7-5bf8-4b04-840c-c3af9ab597cb)

Can we have a task documentation for enabling FIPS compliance in Istio 1.23?

**Describe alternatives you've considered**

I am trying to build the istio/proxy image with *--define boringssl=fips* as mentioned in https://github.com/istio/istio/issues/28476#issuecomment-720300140 and getting below error. Can anyone suggest on building the Istio images with FIPS crypto modules?

![image](https://github.com/user-attachments/assets/f84683ce-038d-44c0-9d4a-5f081219321a)

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[X] Docs
[ ] Dual Stack
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by howardjohn** [comment]
> Is there any specific FIPS compliant image tag we should use along with the env variable?

No, Istio does not publish FIPS builds.

> Can we have a task documentation for enabling FIPS compliance in Istio 1.23?

No, due to above.

**Reply by smvdn** [reply]
@howardjohn Just want to make it clear, we do need to build Istio Proxy (Envoy) Image with --define boringssl=fips along with adding the environment variable COMPLIANCE_POLICY="fips-140-2" to be FIPS Compliant right?

Also, any suggestion on Clang version error during Build? I am using gcr.io/istio-testing/build-tools-proxy:release-1.23-latest container to Build
